core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2029,6 +2029,8 @@ class Log(@volatile private var _dir: File,
           }
           warn(s"Truncated to offset $targetOffset from the log end offset $originalLogEndOffset " +
             s"with ${originalLogEndOffset - targetOffset} messages and $bytesTruncated bytes truncated")
+          LogTruncationStats.logTruncatedBytesRate.mark(bytesTruncated)
+          LogTruncationStats.logTruncatedMessagesRate.mark(originalLogEndOffset - targetOffset)
```

Comment:
Nit: can we unify the logic in line 2031 and 2033 by doing `val messagesTruncated = originalLogEndOffset - targetOffset`?

Reply:
Thanks for the suggestion. I've revised the PR.